data/functions.sql

```diff
@@ -105,8 +108,13 @@ BEGIN
                                     'hostel', 'hotel', 'motel', 'museum')
                    OR railway_val IN ('subway_entrance')) THEN 17
              WHEN (amenity_val IN ('bench', 'waste_basket')) THEN 18
-             ELSE NULL END
-    );
+             ELSE NULL END;
+  RETURN (CASE
+    WHEN way_area > 1.0e7 THEN zoom - 4
+    WHEN way_area > 1.0e6 THEN zoom - 3
+    WHEN way_area > 1.0e5 THEN zoom - 2
+    WHEN way_area > 1.0e4 THEN zoom - 1
+    ELSE zoom END);
```

Comment:
We'll probably want to update the `start_zoom` in queries.yaml to reflect the lowest possible zoom level that we can return here now.